src/test/app/Wasm_test.cpp

```diff
@@ -695,6 +695,1089 @@ struct Wasm_test : public beast::unit_test::suite
+                0x6D,
+                0x01,
+                0x00,
+                0x00,
+                0x00,  // Valid header
+                0x01,
+                0xFF,
+                0xFF,
+                0xFF,
+                0xFF,  // Invalid type section with huge length
+            };
+            auto& engine = WasmEngine::instance();
+
+            auto re = engine.run(malformedSection, "test", {});
+            BEAST_EXPECT(!re.has_value());
+        }
+
+        // Test resource exhaustion through large allocations
+        {
+            auto const wasmStr =
+                boost::algorithm::unhex(allHostFunctionsWasmHex);
+            Bytes const wasm(wasmStr.begin(), wasmStr.end());
+            TestHostFunctions hfs(env);
+
+            // Override getTxField to return extremely large data
+            struct LargeDataHostFunctions : public TestHostFunctions
+            {
+                explicit LargeDataHostFunctions(Env& env)
+                    : TestHostFunctions(env)
+                {
+                }
+
+                Expected<Bytes, HostFunctionError>
+                getTxField(SField const& fname) override
+                {
+                    // Return data that would exceed memory limits
+                    return Bytes(MAX_PAGES * 64 * 1024 + 1, 0xFF);
+                }
+            };
+
+            LargeDataHostFunctions largeHfs(env);
+            auto re = runEscrowWasm(
+                wasm, ESCROW_FUNCTION_NAME, {}, &largeHfs, 100000);
+
+            if (BEAST_EXPECT(re.has_value()))
+            {
+                // Should fail gracefully with error code, not crash
+                BEAST_EXPECTS(re->result == -201, std::to_string(re->result));
+            }
+        }
+    }
+
+    void
+    testWamrAdvancedSecurity()
+    {
+        testcase("WAMR advanced security and attack vectors");
+
+        using namespace test::jtx;
+        Env env(*this);
+
+        // Test module corruption scenarios
+        {
+            // Test WASM with corrupted magic number
+            Bytes corruptedMagic = {
+                0x00,
+                0x61,
+                0x73,
+                0x6D,  // Valid magic
+                0xFF,
+                0x00,
+                0x00,
+                0x00  // Invalid version
+            };
+            auto& engine = WasmEngine::instance();
+            auto re = engine.run(corruptedMagic, "test", {});
+            BEAST_EXPECT(!re.has_value());
+        }
+
+        {
+            // Test WASM with truncated module
+            auto const wasmStr = boost::algorithm::unhex(fibWasmHex);
+            Bytes truncatedWasm(
+                wasmStr.begin(), wasmStr.begin() + 20);  // Truncate
+            auto& engine = WasmEngine::instance();
+
+            auto re = engine.run(truncatedWasm, "fib", wasmParams(5));
+            BEAST_EXPECT(!re.has_value());
+        }
+
+        // Test import/export manipulation attacks
+        {
```

Comment:
This is unnecessary. Such attack is not possible as gas cost doesn't provided by user.  It just duplicate "consume all gas" test